tc_build/llvm.py

```diff
@@ -0,0 +1,545 @@
+        # depends on it. Disable the option to avoid breaking the build, the
+        # kernel does not depend on it.
+        if self.project_is_enabled('compiler-rt') and not Path('/usr/include/execinfo.h').exists():
+            self.cmake_defines['COMPILER_RT_BUILD_GWP_ASAN'] = 'OFF'
+        if self.cmake_defines['CMAKE_BUILD_TYPE'] == 'Release':
+            self.cmake_defines['LLVM_ENABLE_WARNINGS'] = 'OFF'
+        if self.tools.llvm_tblgen:
+            self.cmake_defines['LLVM_TABLEGEN'] = self.tools.llvm_tblgen
+        self.cmake_defines['LLVM_TARGETS_TO_BUILD'] = ';'.join(self.targets)
+        if self.tools.ld:
+            self.cmake_defines['LLVM_USE_LINKER'] = self.tools.ld
+
+        # Clear Linux needs a different target to find all of the C++ header files, otherwise
+        # stage 2+ compiles will fail without this
+        # We figure this out based on the existence of x86_64-generic-linux in the C++ headers path
+        if glob.glob('/usr/include/c++/*/x86_64-generic-linux'):
+            self.cmake_defines['LLVM_HOST_TRIPLE'] = 'x86_64-generic-linux'
+
+        # By default, the Linux triples are for glibc, which might not work on
+        # musl-based systems. If clang is available, get the default target triple
+        # from it so that clang without a '--target' flag always works. This
+        # behavior can be opted out of by setting DISTRIBUTING=1 in the
+        # script's environment, in case the builder intends to distribute the
+        # toolchain, as this may not be portable. Since distribution is not a
+        # primary goal of tc-build, this is not abstracted further.
+        if shutil.which('clang') and not os.environ.get('DISTRIBUTING'):
+            default_target_triple = subprocess.run(['clang', '-print-target-triple'],
+                                                   capture_output=True,
+                                                   check=True,
+                                                   text=True).stdout.strip()
+            self.cmake_defines['LLVM_DEFAULT_TARGET_TRIPLE'] = default_target_triple
+
+        cmake_cmd += [f'-D{key}={self.cmake_defines[key]}' for key in sorted(self.cmake_defines)]
+
+        self.clean_build_folder()
+        self.run_cmd(cmake_cmd)
+
+    def host_target(self):
+        uname_to_llvm = {
+            'aarch64': 'AArch64',
+            'armv7l': 'ARM',
+            'i386': 'X86',
+            'mips': 'Mips',
+            'mips64': 'Mips',
+            'ppc': 'PowerPC',
+            'ppc64': 'PowerPC',
+            'ppc64le': 'PowerPC',
+            'riscv32': 'RISCV',
+            'riscv64': 'RISCV',
+            's390x': 'SystemZ',
+            'x86_64': 'X86',
+        }
+        return uname_to_llvm.get(platform.machine())
+
+    def host_target_is_enabled(self):
+        if 'all' in self.targets:
+            return True
+
+        return self.host_target() in self.targets
```

Comment:
```python
return 'all' in self.targets or self.host_target() in self.targets`
```

Reply:
You do this pattern below in `project_is_enabled`.

Reply:
Ha, good catch! If there is one thing I am good at, it is inconsistency :P

4fd6541ae36f1fdb420c988810dc8b26255c01c2